source/locales.tex

```diff
@@ -4632,3 +4651,717 @@
+    IBMThai = 2016,
+    HPLegal = 2017,
+    HPPiFont = 2018,
+    HPMath8 = 2019,
+    HPPSMath = 2020,
+    HPDesktop = 2021,
+    VenturaMath = 2022,
+    MicrosoftPublishing = 2023,
+    Windows31J = 2024,
+    GB2312 = 2025,
+    Big5 = 2026,
+    Macintosh = 2027,
+    IBM037 = 2028,
+    IBM038 = 2029,
+    IBM273 = 2030,
+    IBM274 = 2031,
+    IBM275 = 2032,
+    IBM277 = 2033,
+    IBM278 = 2034,
+    IBM280 = 2035,
+    IBM281 = 2036,
+    IBM284 = 2037,
+    IBM285 = 2038,
+    IBM290 = 2039,
+    IBM297 = 2040,
+    IBM420 = 2041,
+    IBM423 = 2042,
+    IBM424 = 2043,
+    PC8CodePage437 = 2011,
+    IBM500 = 2044,
+    IBM851 = 2045,
+    PCp852 = 2010,
+    IBM855 = 2046,
+    IBM857 = 2047,
+    IBM860 = 2048,
+    IBM861 = 2049,
+    IBM863 = 2050,
+    IBM864 = 2051,
+    IBM865 = 2052,
+    IBM868 = 2053,
+    IBM869 = 2054,
+    IBM870 = 2055,
+    IBM871 = 2056,
+    IBM880 = 2057,
+    IBM891 = 2058,
+    IBM903 = 2059,
+    IBM904 = 2060,
+    IBM905 = 2061,
+    IBM918 = 2062,
+    IBM1026 = 2063,
+    IBMEBCDICATDE = 2064,
+    EBCDICATDEA = 2065,
+    EBCDICCAFR = 2066,
+    EBCDICDKNO = 2067,
+    EBCDICDKNOA = 2068,
+    EBCDICFISE = 2069,
+    EBCDICFISEA = 2070,
+    EBCDICFR = 2071,
+    EBCDICIT = 2072,
+    EBCDICPT = 2073,
+    EBCDICES = 2074,
+    EBCDICESA = 2075,
+    EBCDICESS = 2076,
+    EBCDICUK = 2077,
+    EBCDICUS = 2078,
+    Unknown8BiT = 2079,
+    Mnemonic = 2080,
+    Mnem = 2081,
+    VISCII = 2082,
+    VIQR = 2083,
+    KOI8R = 2084,
+    HZGB2312 = 2085,
+    IBM866 = 2086,
+    PC775Baltic = 2087,
+    KOI8U = 2088,
+    IBM00858 = 2089,
+    IBM00924 = 2090,
+    IBM01140 = 2091,
+    IBM01141 = 2092,
+    IBM01142 = 2093,
+    IBM01143 = 2094,
+    IBM01144 = 2095,
+    IBM01145 = 2096,
+    IBM01146 = 2097,
+    IBM01147 = 2098,
+    IBM01148 = 2099,
+    IBM01149 = 2100,
+    Big5HKSCS = 2101,
+    IBM1047 = 2102,
+    PTCP154 = 2103,
+    Amiga1251 = 2104,
+    KOI7switched = 2105,
+    BRF = 2106,
+    TSCII = 2107,
+    CP51932 = 2108,
+    windows874 = 2109,
+    windows1250 = 2250,
+    windows1251 = 2251,
+    windows1252 = 2252,
+    windows1253 = 2253,
+    windows1254 = 2254,
+    windows1255 = 2255,
+    windows1256 = 2256,
+    windows1257 = 2257,
+    windows1258 = 2258,
+    TIS620 = 2259,
+    CP50220 = 2260
+  };
+}
+\end{codeblock}
+
+\begin{note}
+The \tcode{text_encoding::id} enumeration
+contains an enumerator for each known registered character encoding.
+For each encoding, the corresponding enumerator is derived from
+the alias beginning with ``\tcode{cs}'', as follows
+\begin{itemize}
+\item
+\tcode{csUnicode} is mapped to \tcode{text_encoding::id::UCS2},
+\item
+\tcode{csIBBM904} is mapped to \tcode{text_encoding::id::IBM904}, and
+\item
+the ``\tcode{cs}'' prefix is removed from other names.
+\end{itemize}
+\end{note}
+
+\rSec3[text.encoding.hash]{Hash specialization}
+
+\indexlibraryglobal{hash}%
+\begin{itemdecl}
+template<class T> struct hash;
```

Comment:
```suggestion
\indexlibrarymember{hash}{text_encoding}%
\begin{itemdecl}
```

Reply:
For the index, don't we want this?
`\indexlibrary{\idxcode{hash}!specializations!\idxcode{text_encoding}}%`

I'll remove the struct in a separate commit since it's in the paper.

Reply:
I don't know.
I just looked up how it's done elsewhere in the LaTeX sources.

Reply:
I've seen it done both ways. Which is preferred and when?

Reply:
I don't think we use `specializations` for any other specialization of `hash`. We seem to use it exclusively for `formatter`.